### PR TITLE
 #0 | voiding form mapping while deleting entity like program, subjec…

### PR DESCRIPTION
**Pull request by swapnil106111**
Voiding form mapping while deleting entity

### CODE REVIEW

File: openchs-server-api/src/main/java/org/openchs/web/EncounterTypeController.java [hunk 1]
```diff
@@ -182,12 +182,19 @@ private FormMapping createFormMapping() {
     @Transactional
     public ResponseEntity voidEncounterType(@PathVariable("id") Long id) {
         OperationalEncounterType operationalEncounterType = operationalEncounterTypeRepository.findOne(id);
+        List<FormMapping> formMappings = formMappingRepository.findAllByIsVoidedFalse();
```
Comment (vinayvenu) [inline openchs-server-api/src/main/java/org/openchs/web/EncounterTypeController.java:185]: You should ideally only be retrieving form mappings that need to be voided, not everything.

File: openchs-server-api/src/main/java/org/openchs/web/EncounterTypeController.java [hunk 2]
```diff
@@ -182,12 +182,19 @@ private FormMapping createFormMapping() {
...
         EncounterType encounterType = operationalEncounterType.getEncounterType();
         if (encounterType == null)
             return ResponseEntity.notFound().build();
 
+        for (int formMapping = 0; formMapping < formMappings.size(); formMapping++) {
+            if(formMappings.get(formMapping).getEncounterTypeUuid() != null && formMappings.get(formMapping).getEncounterTypeUuid().equals(encounterType.getUuid())){
+                formMappings.get(formMapping).setVoided(true);
+            }
+        }
+
```
Comment (vinayvenu) [inline openchs-server-api/src/main/java/org/openchs/web/EncounterTypeController.java:197]: This logic is repeated multiple times in this commit. You should be moving this logic to a common service instead.

**Comment by hithacker** [comment]
@vinayvenu Do we need to keep this open or close it?

**Comment by hithacker** [comment]
Closing this since this is open from a long time and there is no response

**Pull request closed without merging**